### DISCUSSION TITLE
Detecting concept drift using the ASReview Lab sample datatool

### DISCUSSION THREAD

**Discussion by RiannevanderFeen** [asreview/asreview#1880] [Q&A]
I used the sample datatool to collect a random sample of 150 records (old, new and unlabled). But how do I check for concept drift? The Github readme section says nothing about how to go about doing that...

**Comment by Rensvandeschoot** [comment]
Detecting concept drift involves analyzing whether there are changes in the distribution of topics or key terms between older and newer records. Are certain terms or descriptions appearing more often in newer records but not in older ones? This difference could indicate concept drift. It’s up to you as the domain expert to determine which specific terms or concepts are evolving, and whether these changes might reflect concept drift in your data. After screening the set of 150 records you could have an indication whether this is the case.

**Reply by RiannevanderFeen** [reply]
Thank you!

So far I screened 36 (actually 26) of the 150 records again in ASReview Lab (it took 5 relevant and 5 irrelevant automatically from the labeled list) and I noticed that it takes a long time for me to get relevant records (on average 1 in every 10 - 14 records). But the thing is that I already screened up to the stopping points (phase 2) in the dataset where I got this batch of 150 from.

What conclusions can I draw from this? That it is less and less likely for me to find relevant records in the remaining original dataset?
And what about the concept drift? How can I look at the descriptions and terms myself? Are those found in one of the columns in the Excel table? (Sorry for my confusion).  

I actually have another import question in relation to the scrutiny of my labelling etc.: I noticed that the 2*(FRR_t x T) is much larger (twice as large) than the 10% of T. So I went back to check my first random batch of relevant records in the 0.01% of T. To my surprise I had been pretty enthusiastic with including papers that do not match in inclusion criteria previously drawn up. When subtracting those wrongly labeled, the FRR_t is a much better fit to the 10%. 

In this case I am talking about 26 relevant records in 276 (0,01%) screened records. Of those, 15 turned out to be wrongly included for different reasons. In other words, 11 were rightly included as prior knowledge to warm up the AI. Is it fair to say that this is still a decent amount and on the whole, the scale of records being screened in phase 2 (2,737 records), it is a reliable amount to have trained the active learning model properly?

Note: It’s a good recommendation for reviewers using this random step to not get too exited to start reviewing once you’ve warmed up the AI. To take a break and first go back and check the labelled records. Perhaps to start over if you find that you’ve included many that should have been excluded in hindsight? At the same time, perhaps it’s part of an active learning process for the human intelligence as well, to have this ‘concept convergence’ happening while screening so many records, when at the time you started, you might have been adrift 😉

**Reply by Rensvandeschoot** [reply]
## Importance of Initial Preparation

I really appreciate your recommendation about taking a break before starting the main screening phase. Before diving into screening, it’s important to take the time to become the "oracle" of the process—the expert on the topic—so you’re in control and confidently guiding the AI on what’s relevant. Rushing through this phase can lead to inconsistencies, as you noted.

## Screening the Batch of 150 Records

Regarding the batch of 150 records, it makes sense that you find only a small number of relevant records. In a sample of this size, especially with the skewed distribution you described, there will likely only be a few relevant papers, if any. This step serves more as an opportunity for you to become familiar with the dataset and get a sense of how abstracts in older and newer papers might differ. In some of my own projects, we screened this batch entirely as a team to compare notes and ensure consistency, and then extracted another subset of 150 records to independently assess. Any disagreements were discussed, and we adjusted the logbook with our inclusion criteria accordingly. Only then did we proceed to the main screening phase, using both subsets (150+150) as training data.

## Investigating Concept Drift

To investigate concept drift, you can compare abstracts in older and newer records to look for shifts in terminology or key concepts. These abstracts are indeed found in one of the columns of your Excel dataset, typically titled "abstract" or something similar. Comparing them systematically will help you understand any evolving language or themes over time.

## Evaluating Inclusion Criteria and FRR\_t

Regarding your question about the FRR\_t being much larger, it’s good that you revisited and reassessed the records you initially included. It’s entirely normal for inclusion criteria to evolve as you gain a better understanding of the dataset and the topic. With the adjustments you made, having 11 correctly included records is still a decent number, and it’s likely sufficient to effectively train the active learning model, especially considering the scale of phase 2.

**Reply by RiannevanderFeen** [reply]
Thanks so much for the reasurance and detailed explainations. I guess I have other expectations of the concept drift check. It sounds like this is a manual (not AI) driven aspect. Perhaps a request for the future (but I believe I've asked that in the summer course already), to have a 'search term network drawing' (or word cloud, mentimeter) or something, that gives the screener feedback on terms that seem dominant/prevelant.

**Reply by RiannevanderFeen** [reply]
Actually, because you are connecting it to publication date, it seems you are interested in the evolution of concepts over time in the literature. Is that what you are referring to with 'concept drift'?  

I thought what was meant was a risk in the labelling process itself (so intra-reviewer reliability, if that even exists). Or is it simply both? 

In our research it's a given that there are many terms used to describe the same thing and I don't see the use of knowing what student-student interaction, for example, could have been referred to in 1989...as it was probably still called something like peer support or whatever.

Still, this is a good reason of why to use ASReview lab and huge amounts of data, exactly because there are so many synonyms in the area of social behavior science.    

Perhaps you have a good example of the link between publication date and concept drift?

**Comment by Rensvandeschoot** [comment]
> Thanks so much for the reasurance and detailed explainations. I guess I have other expectations of the concept drift check. It sounds like this is a manual (not AI) driven aspect. Perhaps a request for the future (but I believe I've asked that in the summer course already), to have a 'search term network drawing' (or word cloud, mentimeter) or something, that gives the screener feedback on terms that seem dominant/prevelant.


Yes, it is a human process, and a word cloud is a good idea. You could create two datasets (using reference manager software): one with the oldest records and one with the newest, and then create a word cloud. You can use our tool here: [ASReview Wordcloud](https://github.com/asreview/asreview-wordcloud?tab=readme-ov-file#asreview-wordcloud), or any other word cloud tool.

Although this process is largely manual, there are many ways you could expand it into a data science project. For example, you could use natural language processing (NLP) techniques to automatically track the evolution of key terms or visualize these changes using more sophisticated data analysis tools.

**Reply by RiannevanderFeen** [reply]
@Rensvandeschoot by the way, this is the Wordcloud I got using ASReview Wordcloud:

![Wordcloud relevant student_student network](https://github.com/user-attachments/assets/2ec6df64-a323-496b-87ae-e23fa50aaa1f)

It helps me verify my concept of inclusion criteria. But I just have to say, this is really awesome stuff!!

**Comment by Rensvandeschoot** [comment]
> Actually, because you are connecting it to publication date, it seems you are interested in the evolution of concepts over time in the literature. Is that what you are referring to with 'concept drift'?

Concept drift can be caused by many factors, including time. It is a separate area of research, and I’m not an expert in this field. However, it is relevant for ASReview because if you select a feature extractor like TF-IDF, it assumes that every abstract uses the same vocabulary consistently. If this is not the case—if terms have changed their meaning over time, or vary across continents, cultures, etc.—then you might achieve better performance with transformer models (such as sBERT) that can capture context more effectively.

**Reply by RiannevanderFeen** [reply]
Excellent! That is just what I needed to hear. I had trouble downloading and/or using sBert, but that's another thread and has no rush at the moment. Thanks for all your assistance!
